content-signup/src/content-signup.js

```diff
@@ -1,105 +1,121 @@
+  var name = responses.Name[0].trim();
+  var topicsString = responses.Topics[0].toLowerCase();
 
-    // If there is at least one topic selected, send email to the recipient,
-    // with this subject and email body.
-    if (topics.length > 0) {
-      let emailBody = createEmailBody(recipient, topics);
-      MailApp.sendEmail({
-        to: recipient[EMAIL],
-        subject: "Howdy",
-        htmlBody: emailBody,
-      });
+  // Parse topics of interest into a list (since there are multiple items
+  // that are saved in the row as blob of text).
+  var topics = [];
+  for (var topic in topicUrls) {
+    // indexOf searches for the topic in topicsString and returns a non-negative
+    // index if the topic is found, or it will return -1 if it's not found.
+    if (topicsString.indexOf(topic.toLowerCase()) != -1) {
+      topics.push(topic);
     }
-    
-    // Write the email sent date on the 'F' column.
-    let now = new Date();
-    sheet.getRange(`F${recipient.rowNum+1}`).setValue(now.toString());
+  }
 
-    // Logs activity.
-    Logger.log(JSON.stringify(recipient));
+  // If there is at least one topic selected, send an email to the recipient.
+  var status = '';
+  if (topics.length > 0) {
+    MailApp.sendEmail({
+      to: email,
+      subject: emailSubject,
+      htmlBody: createEmailBody(name, topics),
+    });
+    status = 'Sent';
   }
+  else {
+    status = 'No topics selected';
+  }
+
+  // Append the status on the spreadsheet to the responses' row.
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var row = e.range.getRow();
+  var column = e.values.length + 1;
+  sheet.getRange(row, column).setValue(status);
+
+  // Log activity.
+  Logger.log("status=" + status + "; responses=" + JSON.stringify(responses));
 }
 
-// Creates email body and includes the links based on topic.
-function createEmailBody(recipient, topics) {
-  let topicsHTML = '<ul>\n';
-  for (let topic of topics) {
-    let url = TOPIC_URLS[topic];
-    topicsHTML += `<li><a href="${url}">${topic}</a></li>\n`;
+/**
+ * Creates email body and includes the links based on topic.
+ *
+ * @param {string} recipient - The recipient's email address.
+ * @param {string[]} topics - List of topics to include in the email body.
+ * @return {string} The email body as an HTML string.
+ */
+function createEmailBody(name, topics) {
+  var topicsHTML = '<ul>\n';
+  for (var i in topics) {
+    var topic = topics[i];
+    var url = topicUrls[topic];
+    topicsHTML += '<li><a href="' + url + '">' + topic + '</a></li>\n';
   }
   topicsHTML += '</ul>';
   
-  // Make sure to update the EMAIL_TEMPLATE_DOC_ID at the top.
-  let emailBody = docToHtml(EMAIL_TEMPLATE_DOC_ID);
-  Logger.log(JSON.stringify(emailBody));
-  emailBody = emailBody.replace('{{NAME}}', recipient[NAME]);
+  // Make sure to update the emailTemplateDocId at the top.
+  var emailBody = docToHtml(emailTemplateDocId);
+  emailBody = emailBody.replace('{{NAME}}', name);
```

Comment:
If you want to support having the `{{NAME}}` placeholder more than once in the doc, then you need to use a regex instead, with the `/g` option:

```
emailBody.replace(/{{NAME}}/g, 'Eric');
```